hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerRpcConnection.java

```diff
@@ -366,13 +366,16 @@ public void processOneRpc(ByteBuff buf) throws IOException, InterruptedException
       processConnectionHeader(buf);
       callCleanupIfNeeded();
       this.connectionHeaderRead = true;
+      this.rpcServer.getRpcCoprocessorHost().preAuthorizeConnection();
```

Comment:
i wonder if there's anything else we can send in here? We have the RequestHeader at this point, and maybe the IP address?